rotkehlchen/chain/bitcoin/manager.py

```diff
@@ -0,0 +1,213 @@
+            )
+            balances[account] = (balance := funded_txo_sum - spent_txo_sum)
+            have_transactions[account] = ((stats['tx_count'] != 0), balance)
+
+        return balances if action == 'balances' else have_transactions
+
+    @overload
+    def _query_blockchain_info(
+            self,
+            action: Literal['balances'],
+            accounts: Sequence[BTCAddress],
+    ) -> dict[BTCAddress, FVal]:
+        ...
+
+    @overload
+    def _query_blockchain_info(
+            self,
+            action: Literal['has_transactions'],
+            accounts: Sequence[BTCAddress],
+    ) -> dict[BTCAddress, tuple[bool, FVal]]:
+        ...
+
+    def _query_blockchain_info(
+            self,
+            action: Literal['balances', 'has_transactions'],
+            accounts: Sequence[BTCAddress],
+    ) -> dict[BTCAddress, FVal] | dict[BTCAddress, tuple[bool, FVal]]:
+        """Queries blockchain.info
+        May raise:
+        - RemoteError if got problems with querying the API
+        - UnableToDecryptRemoteData if unable to load json in request_get
+        - KeyError if got unexpected json structure
+        - DeserializationError if got unexpected json values
+        """
+        results: list[dict[str, Any]] = []
+        accounts_chunks = [accounts[x:x + 80] for x in range(0, len(accounts), 80)]
+        for accounts_chunk in accounts_chunks:
+            params = '|'.join(accounts_chunk)
+            btc_resp = request_get_dict(
+                url=f'https://blockchain.info/multiaddr?active={params}',
+                handle_429=True,
+                # If we get a 429 then their docs suggest 10 seconds (https://blockchain.info/q)
+                backoff_in_seconds=10,
+            )
+            results.extend(btc_resp['addresses'])
+
+        if action == 'balances':
+            balances: dict[BTCAddress, FVal] = {}
+            for entry in results:
+                balances[entry['address']] = satoshis_to_btc(
+                    ensure_type(
+                        symbol=entry['final_balance'],
+                        expected_type=int,
+                        location='blockchain.info "final_balance"',
+                    ),
+                )
+            return balances
+        else:  # action == 'has_transactions'
+            have_transactions = {}
+            for entry in results:
+                balance = satoshis_to_btc(entry['final_balance'])
+                have_transactions[entry['address']] = (entry['n_tx'] != 0, balance)
+            return have_transactions
+
+    @overload
+    def _query(
+            self,
+            action: Literal['balances'],
+            accounts: Sequence[BTCAddress],
+    ) -> dict[BTCAddress, FVal]:
+        ...
+
+    @overload
+    def _query(
+            self,
+            action: Literal['has_transactions'],
+            accounts: Sequence[BTCAddress],
+    ) -> dict[BTCAddress, tuple[bool, FVal]]:
+        ...
+
+    def _query(
```

Comment:
There is no reason for a single function with union type returns with if. Keep it simpler and lose that if check and the union type

```
  Or even simpler - keep the public methods but make the internal helper methods specific:
  def get_balances(self, accounts: Sequence[BTCAddress]) -> dict[BTCAddress, FVal]:
      return self._query_for_balances(accounts)

  def have_transactions(self, accounts: list[BTCAddress]) -> dict[BTCAddress, tuple[bool, FVal]]:
      return self._query_for_transactions(accounts)
```